pyomo/contrib/incidence_analysis/interface.py

```diff
@@ -101,10 +111,36 @@ def get_bipartite_incidence_graph(variables, constraints, include_fixed=True):
     graph.add_nodes_from(range(M), bipartite=0)
     graph.add_nodes_from(range(M, M + N), bipartite=1)
     var_node_map = ComponentMap((v, M + i) for i, v in enumerate(variables))
-    for i, con in enumerate(constraints):
-        for var in identify_variables(con.expr, include_fixed=include_fixed):
-            if var in var_node_map:
-                graph.add_edge(i, var_node_map[var])
+    vars_in_cons = list(
+        _generate_variables_in_constraints(constraints, include_fixed=True)
+    )
+    to_reset = [
+        var for var in vars_in_cons if var.fixed and var.value is None
+    ]
```

Comment:
Fixed variables with values of `None` cause problems when we attempt to evaluate them to deterime the constant portion or linear coefficients of the expression. However, the lack of a value for these cofficients/constants should not prevent us from constructing the incidence graph, so we will set them to something other than `None` to generate the `AMPLRepn`, then reset them to `None` afterwards.

Reply:
I think `generate_standard_repn` has a `compute_values=False` option which allows for things like this. Does the AMPLRepn have a similar concept?

Reply:
Does `generate_standard_repn` do "linear cancellations", including `0*x -> 0`, with `compute_values=False`?

Reply:
where `x` is unfixed or fixed to `None`?

Reply:
Maybe it does not matter.

Reply:
I specifically would like it to make this optimization when `x` is unfixed.

Reply:
If `y` is fixed to zero and `compute_values=False`, I would like it to *not* make the optimization `x*y -> 0`. I'll test this behavior and prototype a `generate_standard_repn` backend when I have some time.

Reply:
I think this is the behavior you are looking for?

```
In [1]: import pyomo.environ as pe

In [2]: from pyomo.repn.standard_repn import generate_standard_repn

In [3]: from pyomo.core.expr.numeric_expr import ProductExpression

In [4]: m = pe.ConcreteModel()

In [5]: m.x = pe.Var()

In [6]: print(generate_standard_repn(ProductExpression((0, m.x)), compute_values=False))

constant:       0
linear vars:    []
linear var ids: []
linear coef:    []
quadratic vars:    []
quadratic var ids: []
quadratic coef:    []
nonlinear expr: None
nonlinear vars: []



In [7]: m.y = pe.Var()

In [8]: m.y.fix(0)

In [9]: print(generate_standard_repn(m.x*m.y, compute_values=False))

constant:       0
linear vars:    ['x']
linear var ids: [4377918416]
linear coef:    [<pyomo.core.base.var.ScalarVar object at 0x1065c59d0>]
quadratic vars:    []
quadratic var ids: []
quadratic coef:    []
nonlinear expr: None
nonlinear vars: []
```